Tasks/AzureRmWebAppDeployment/xdtutility.ts

```diff
@@ -0,0 +1,48 @@
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
+
+function applyXmlTransformation(sourceFile, transformFile) {
+	var cttPath = path.join(__dirname, 'ctt/ctt.exe');
+    var cttArgs = ' s:"' + sourceFile + '" t:"' + transformFile + '" d:"' + sourceFile + '" pw';
+    var cttExecutionResult = tl.execSync(cttPath, cttArgs, <trm.IExecOptions>{failOnStdErr: true, silent: true});
+    if(cttExecutionResult.stderr)
+        throw "XML Transformation Error while transforming " + sourceFile + " using " + transformFile;
+}
+
+export function xmlTransform(transformXmlPattern : string, sourceXmlPattern : string) {
```

Comment:
The stack trace of the error is displaced on the console by ctt.exe. So not displaying it again in the function.